sphinx_gallery/docs_resolv.py

```diff
@@ -232,22 +234,34 @@ def _get_link(self, cobj):
                 fname_idx = value[cobj['name']][0]
 
         if fname_idx is not None:
-            fname = self._searchindex['filenames'][fname_idx] + '.html'
-
-            if self._is_windows:
-                fname = fname.replace('/', '\\')
-                link = os.path.join(self.doc_url, fname)
-            else:
-                link = posixpath.join(self.doc_url, fname)
-
-            if hasattr(link, 'decode'):
-                link = link.decode('utf-8', 'replace')
-
-            if link in self._page_cache:
-                html = self._page_cache[link]
+            fname = self._searchindex['filenames'][fname_idx]
+            # In 1.5+ Sphinx seems to have changed from .rst.html to only
+            # .html extension in converted files. But URLs could be
+            # built with < 1.5 or >= 1.5 regardless of what we're currently
+            # building with, so let's just check both :(
+            fnames = [fname + '.html', os.path.splitext(fname)[0] + '.html']
+            for fname in fnames:
+                try:
+                    if self._is_windows:
+                        fname = fname.replace('/', '\\')
+                        link = os.path.join(self.doc_url, fname)
+                    else:
+                        link = posixpath.join(self.doc_url, fname)
+
+                    if hasattr(link, 'decode'):
+                        link = link.decode('utf-8', 'replace')
+
+                    if link in self._page_cache:
+                        html = self._page_cache[link]
+                    else:
+                        html = get_data(link, self.gallery_dir)
+                        self._page_cache[link] = html
+                except Exception as e:
+                    pass
+                else:
+                    break
             else:
-                html = get_data(link, self.gallery_dir)
-                self._page_cache[link] = html
+                raise e
```

Comment:
Would `raise` not be work to raise the last caught exception? My experience is that `raise e` creates a misleading traceback especially on Python 2.7 since you lose the provenance of the original exception.